python/paddle/fluid/tests/unittests/test_splits_api.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import numpy as np
+import paddle
+import paddle.fluid as fluid
+import paddle.fluid.core as core
+
+
+def func_ref(func, x, num_or_sections):
+    # Convert the num_or_sections in paddle to indices_or_sections in numpy
+    # Do not support -1
+    if isinstance(num_or_sections, int):
+        indices_or_sections = num_or_sections
+    else:
+        indices_or_sections = np.cumsum(num_or_sections)[:-1]
+    return func(x, indices_or_sections)
+
+
+# TODO: add other split API, such as dsplit、hsplit
+test_list = [
+    (paddle.vsplit, np.vsplit),
+]
+
+
+class TestSplitsAPI(unittest.TestCase):
+
+    def setUp(self):
+        self.shape = [4, 5, 2]
+        self.num_or_sections = 2
+        self.x_np = np.random.uniform(-1, 1, self.shape).astype('float64')
+        self.place = paddle.CUDAPlace(0) if core.is_compiled_with_cuda() \
+            else paddle.CPUPlace()
+
+    def test_static_api(self):
+        paddle.enable_static()
+        for func, func_type in test_list:
+            with paddle.static.program_guard(paddle.static.Program()):
+                x = paddle.fluid.data('X', self.x_np.shape, self.x_np.dtype)
+                out = func(x, self.num_or_sections)
+                exe = paddle.static.Executor(self.place)
+                res = exe.run(feed={'X': self.x_np}, fetch_list=[out])
+            out_ref = func_ref(func_type, self.x_np, self.num_or_sections)
+            for n, p in zip(out_ref, res):
+                self.assertTrue(np.allclose(n, p))
```

Comment:
<img width="865" alt="image" src="https://user-images.githubusercontent.com/38436475/187272822-98cfb383-50cb-47ef-9432-f69ad50cef8d.png">

使用 `np.testing` 模块下的断言函数代替 `self.assertTrue(np.allclose(...))`

```suggestion
                np.testing.assert_allclose(n, p)
```

如果由于精度问题导致测试无法通过可适当调整 atol 和 rtol，66 行同

修改后应可通过 PR-CI-APPROVAL

Reply:
down